test/hooks.test.js

```diff
@@ -502,6 +502,22 @@ test('onRoute hook should preserve system route configuration', t => {
   })
 })
 
+test('onRoute hook should preserve handler function in options of shorthand route system configuration', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  fastify.register((instance, opts, next) => {
+    instance.addHook('onRoute', function (route) {
+      t.strictEqual(route.handler.length, 2)
```

Comment:
This doesn't test that it's the right `handler` (it just tests that it is some function with a length of 2). The `handler` should be created outside of the options object so that the comparison can be:

```js
t.strictEqual(route.handler, handler)
```

Reply:
@nwoltman I updated the tests, see the code again please.